dds/DCPS/XTypes/DynamicDataBase.cpp

```diff
@@ -243,8 +235,6 @@ DDS::ReturnCode_t DynamicDataBase::check_member(
   bool invalid_tk = true;
   if (is_basic(cmp_type_kind)) {
     invalid_tk = cmp_type_kind != tk;
-  } else if (tk == TK_NONE) {
-    invalid_tk = !is_complex(type_kind);
```

Comment:
Why was this taken out?

Reply:
There were some uses of this function that relied on `tk` taking a default parameter value `TK_NONE`. But those uses are removed and we don't need this else block anymore.

Reply:
In the master branch, `check_member` is called in `get_complex_value` to check for non-complex types. But that is not necessary since we can `get_complex_value` for any member type and get a `DynamicData` for that member.